src/test/AutoRoller.t.sol

```diff
@@ -936,12 +978,195 @@ contract AutoRollerTest is DSTestPlus {
+        // where `if (assets - targetToJoin > 0)` is false.
+        // I think can never happen since `onSponsorWindowOpened`, when calling `_space.getEQReserves`
+        // we are doing `targetedRate < 0.01e18 ? 0.01e18 : targetedRate` which means that we are always
+        // getting a `eqPTReserves` > 0 so we will be issuing PTs. 
+    }
+
+    function testPreviewWithdrawInsufficentLiquidity(uint256 amount) public {
+        // 1. Roll into the first Series.
+        autoRoller.roll();
+
+        // 2. Make a deposit.
+        autoRoller.deposit(1e18, address(this));
+
+        // 3. Expect revert because of InsufficientLiquidity.
+        vm.expectRevert(abi.encodeWithSelector(AutoRoller.InsufficientLiquidity.selector));
+        autoRoller.previewWithdraw(100e18);
+    }
+
+    function testMaxWithdrawDuringCooldown(uint256 amount) public {
+        // 1. Make a deposit durring cooldown.
+        autoRoller.deposit(1e18, address(this));
+
+        // 2. Assert max withdraw is same as deposit.
+        uint256 maxWithdraw = autoRoller.maxWithdraw(address(this));
+        assertEq(maxWithdraw, 1e18);
+    }
+
+    function testRedeemWithYTsExcess() public {
+        // 1. Deposit during the initial cooldown phase.
+        autoRoller.deposit(0.2e18, address(this));
+        assertEq(autoRoller.balanceOf(address(this)), 0.2e18);
+
+        // 3. Roll the Target into the first Series.
+        autoRoller.roll();
+
+        // 4. Deposit during the first active phase.
+        autoRoller.deposit(0.3e18, address(this));
+        assertRelApproxEq(autoRoller.balanceOf(address(this)), 0.5e18, 0.0001e18 /* 0.01% */);
+
+        // 5. Redeem all shares while still in the active phase.
+        uint256 targetBalPre = target.balanceOf(address(this));
+        vm.expectCall(address(periphery), abi.encodeWithSelector(periphery.swapYTsForTarget.selector));
+        autoRoller.redeem(autoRoller.balanceOf(address(this)), address(this), address(this));
+        uint256 targetBalPost = target.balanceOf(address(this));
+        // assertRelApproxEq(targetBalPost - targetBalPre, 0.5e18, 0.0001e18 /* 0.01% */); // TODO: what should it be the expected value?
+
+        // Check that the Target dust leftover is small
+        assertLt(target.balanceOf(address(autoRoller)), 1e9);
+    }
+
+    function testMaxRedeemDuringCooldown(uint256 amount) public {
+        // 1. Make a deposit during cooldown.
+        uint256 sharesOut = autoRoller.deposit(0.1e18, address(this));
+
+        // 2. Assert max withdraw is same sharesOut.
+        assertEq(autoRoller.maxRedeem(address(this)), sharesOut);
+
+        // 4. Roll into the first Series.
+        autoRoller.roll();
+
+        // 3. Make another deposit during cooldown.
+        sharesOut += autoRoller.deposit(0.1e18, address(this));
+
+        vm.warp(autoRoller.maturity());
+        
+        // 4. Increase scale and settle the first Series.
```

Comment:
Did we want to manually increase scale somewhere here?

Reply:
no no, it's not needed